libs/stringCore/include/stringCore/stringReader.h

```diff
@@ -0,0 +1,146 @@
+/**
+  StringReader is a simple operation-based class for pulling data out of strings.
+  TryRead accepts an arbitrary number of operations to perform.
+*/
+#pragma once
```

Comment:
Please make sure that all header files in the public include folder have the `#ifndef` guards. See other header files: they follow the same scheme.